From 89c491bc9746ad7b4804c6868e5ebf360fbe6808 Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Tue, 9 Apr 2013 09:41:07 -0700
Subject: Dalvik: Fix The Registerization Sorting

BZ: 100558

Fixes the sorting comparator in the middle-end registerization algorithm

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-ME
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ifbfc38e2857ac9363201e6dfebc48a29f58e5410
Orig-MCG-Change-Id: I5a5803d971ba60ca5ff76d9686b32d647cbf4893
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Qiming Shi <qiming.shi@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/RegisterizationME.cpp |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/RegisterizationME.cpp b/vm/compiler/RegisterizationME.cpp
index 54793c3..c2581e6 100644
--- a/vm/compiler/RegisterizationME.cpp
+++ b/vm/compiler/RegisterizationME.cpp
@@ -342,7 +342,7 @@ class RegisterSorter
          */
         bool operator() (const std::pair<int, RegisterClass> &first, const std::pair<int, RegisterClass> &second)
         {
-            int firstCount = registerCounts[first.first];
+            int firstCount = registerCounts[first.second];
             int secondCount = registerCounts[second.second];
 
             //We only care about the first
-- 
1.7.4.1

